src/index.js

```diff
@@ -10,20 +10,22 @@ const GoMulticastDNS = require('./compat')
 class MulticastDNS extends EventEmitter {
   constructor (options = {}) {
     super()
-    if (!options.peerInfo) {
-      throw new Error('needs a PeerInfo to work')
+
+    if (!options.peerId) {
+      throw new Error('needs own PeerId to work')
     }
 
     this.broadcast = options.broadcast !== false
     this.interval = options.interval || (1e3 * 10)
     this.serviceTag = options.serviceTag || 'ipfs.local'
     this.port = options.port || 5353
-    this.peerInfo = options.peerInfo
+    this.peerId = options.peerId
+    this.peerMultiaddrs = options.multiaddrs || []
     this._queryInterval = null
     this._onPeer = this._onPeer.bind(this)
 
     if (options.compat !== false) {
-      this._goMdns = new GoMulticastDNS(options.peerInfo, {
+      this._goMdns = new GoMulticastDNS(options.peerId, this.peerMultiaddrs, {
         queryPeriod: options.compatQueryPeriod,
         queryInterval: options.compatQueryInterval
       })
```

Comment:
Existing bug, we're apparently not passing these options into the querier `GoMulticastDNS` creates. It's constructor doesn't take options but it should

Reply:
I will fix this